portality/crosswalks/article_crossref_xml.py

```diff
@@ -0,0 +1,304 @@
+import os
+
+from portality.core import app
+from lxml import etree
+import re
+from portality.bll import exceptions
+from portality.crosswalks.exceptions import CrosswalkException
+from portality import models
+
+NS = {'x': 'http://www.crossref.org/schema/4.4.2', 'j': 'http://www.ncbi.nlm.nih.gov/JATS1'}
+
+
+class CrossrefXWalk(object):
+    format_name = "crossref"
+    schema_path = app.config.get("SCHEMAS", {}).get("crossref")
+
+    def __init__(self):
+        self.validation_log = ""
+
+        # load the schema into memory for more efficient usage in repeat calls to the crosswalk
+        if self.schema_path is None:
+            raise exceptions.IngestException(
+                message="Unable to validate for CrossrefXWalk, as schema path is not set in config")
+
+        try:
+            schema_file = open(self.schema_path)
```

Comment:
Just thinking out-loud, what happens here if the remotely hosted bits of the XSD aren't available for whatever reason (like the mathml.xsd can't be reached because of a network issue that's theirs not ours)?  I wonder if there's a way we can localise or cache those resources so that we don't have to go out and get them every time?

Reply:
We can think about that but these are quite chained (eg mathml.xsd requires other remote resources). We would have to find a way to periodically ensure we have the newest version of these.

Reply:
Yes, so that's an interesting thought - while the Crossref schema remains fixed at a given version, the transient dependencies can change without us directly knowing?  That sounds like a good argument for taking our own copies, so the upload is stable until we decide for ourselves to do an upgrade.  But it also sounds like a pain, so difficult to know what the right answer is.